File: Algorithm/baekjoon/java/src/swea_3282.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class swea_3282 {
    public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st ;

    st = new StringTokenizer(br.readLine());
    int t = Integer.parseInt(st.nextToken());

    for (int tc=1; tc<=t; tc++) {
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        int[][] DP = new int[N+1][K+1];
        int[][] things = new int[N+1][2] ;

        for (int thing=1; thing < N; thing++) {
            st = new StringTokenizer(br.readLine());
            things[thing][0] = Integer.parseInt(st.nextToken());
            things[thing][1] = Integer.parseInt(st.nextToken());
        }

        for (int i=1; i<N+1; i++) {
            for (int j=1; j<K+1; j++) {

                if (j < things[i][0]) {
                    DP[i][j] = DP[i-1][j] ;
                } else {
                    DP[i][j] = Math.max(DP[i-1][j-things[i][0]]+things[i][1], DP[i-1][j] );
                }
            }
        }
        System.out.println("#" + tc + " " + DP[N][K]);
    }



    }
}

File: Algorithm/swea/swea_3282.py
for tc in range(int(input())):
    # 물건 개수 N 
    # 가방 크기 K
    N,K = map(int,input().split())
    
    DP = [ [0] * (K+1) for _ in range(N+1) ]
    things =[[0,0]] + [ [*map(int,input().split())] for _ in range(N)]

    for i in range(N+1):
        for j in range(K+1):

            if j < things[i][0]:
                DP[i][j] = DP[i-1][j]
            else:
                DP[i][j] = max(DP[i-1][j-things[i][0]]+things[i][1], DP[i-1][j])
    print(f"#{tc+1} {DP[N][K]}")